### PR TITLE
network: ndisc: introduce UseGateway= and UseRoutePrefix=

### LINKED ISSUE #21263: [systemd-networkd] UseGateway=no equivalent for IPv6 RA
**Is your feature request related to a problem? Please describe.**
When I have servers with multiple NICs that are dynamically addressed (DHCPv6 or SLAAC for IPv6, DHCPv4 for IPv4), I often want to only have an IP on a certain subnet, without a default route, even if the default gateway sends one.

**Describe the solution you'd like**
On IPv4, we can set UseGateway=false on DHCPv4 to prevent default route from being added.
I don't see such option for IPv6 (either for SLAAC or DHCPv6).

I'd like the following:
```
[IPv6AcceptRA]
UseGateway=false
```

**Describe alternatives you've considered**
I tried the following thing:
```
[IPv6AcceptRA]
RouteDenyList=::/0
```
But it did not work.

**The systemd version you checked that didn't have the feature you are asking for**
systemd 249 (249.5-2-arch)

### PR DESCRIPTION
**Pull request by yuwata**
Closes #21263.

### CODE REVIEW

File: src/network/networkd-ndisc.c [hunk 1]
```diff
@@ -392,6 +392,9 @@ static int ndisc_router_process_autonomous_prefix(Link *link, sd_ndisc_router *r
         assert(link);
         assert(rt);
 
+        if (!link->network->ipv6_accept_ra_use_autonomous_prefix)
```
Comment (bluca) [inline src/network/networkd-ndisc.c:406]: check `link->network` first or assert, same below

File: man/systemd.network.xml [hunk 2]
```diff
@@ -2295,6 +2295,22 @@ Token=prefixstable:2002:da8:1::</programlisting></para>
           </listitem>
         </varlistentry>
 
+        <varlistentry>
+          <term><varname>UseGateway=</varname></term>
+          <listitem>
+            <para>When true (the default), the default gateway with the router address will be
```
Comment (bluca) [inline man/systemd.network.xml]: "the router address will be configured as the default gateway" maybe?

**Review by bluca — APPROVED** [review]
looks good, just two minor things - phrasing in manpages and asserts/checks before dereferencing

**Comment by yuwata** [comment]
@bluca Thank you for your review. The above points are addressed. Upgrading the green label.

**Pull request merged**